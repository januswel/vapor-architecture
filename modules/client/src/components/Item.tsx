import React from 'react'
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native'

const styles = StyleSheet.create({
  container: {
    width: 168,
    height: 168,
    margin: 16,
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  name: {
    fontSize: 24,
  },
  price: {
    fontSize: 16,
  },
})

const soldOutStyle = {
  backgroundColor: 'red',
}
const tooFewChargedMoneyStyle = {
  backgroundColor: 'gray',
}

const ButtonLabel = props => {
  const backgroundStyle = props.isSoldOut ? soldOutStyle : props.isTooFewChargedMoney ? tooFewChargedMoneyStyle : {}

  return (
    <View style={[styles.container, backgroundStyle]}>
      <Text style={styles.name}>{props.name}</Text>
      <Text style={styles.price}>Â¥{props.price}</Text>
    </View>
  )
}

export default props => {
  const isSoldOut = props.item.remains === 0
  const isTooFewChargedMoney = props.vendingMachine.chargedMoney < props.item.drink.price

  return isSoldOut || isTooFewChargedMoney ? (
    <ButtonLabel
      isSoldOut={isSoldOut}
      isTooFewChargedMoney={isTooFewChargedMoney}
      name={props.item.drink.name}
      price={props.item.drink.price}
    />
  ) : (
    <TouchableOpacity
      onPress={() => {
        props.actions.buy(props.index)
      }}
    >
      <ButtonLabel
        isSoldOut={isSoldOut}
        isTooFewChargedMoney={isTooFewChargedMoney}
        name={props.item.drink.name}
        price={props.item.drink.price}
      />
    </TouchableOpacity>
  )
}
