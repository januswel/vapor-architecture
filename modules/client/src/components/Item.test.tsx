import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Item from './Item'

describe('Item', () => {
  it('renders tappable item for available item', () => {
    expect(
      renderer.create(
        <Item
          id={0}
          price={120}
          imageUrl="https://example.com/assets/images/coffee.png"
          isSoldOut={false}
          isTooFewChargedMoney={false}
          actions={{
            sellAndCount: () => {},
          }}
        />,
      ),
    ).toMatchSnapshot()
  })

  it('renders untappable item for unavailable item', () => {
    expect(
      renderer.create(
        <Item
          id={0}
          price={120}
          imageUrl="https://example.com/assets/images/coffee.png"
          isSoldOut
          isTooFewChargedMoney
          actions={{
            sellAndCount: () => {},
          }}
        />,
      ),
    ).toMatchSnapshot()

    expect(
      renderer.create(
        <Item
          id={0}
          price={120}
          imageUrl="https://example.com/assets/images/coffee.png"
          isSoldOut
          isTooFewChargedMoney={false}
          actions={{
            sellAndCount: () => {},
          }}
        />,
      ),
    ).toMatchSnapshot()

    expect(
      renderer.create(
        <Item
          id={0}
          price={120}
          imageUrl="https://example.com/assets/images/coffee.png"
          isSoldOut={false}
          isTooFewChargedMoney
          actions={{
            sellAndCount: () => {},
          }}
        />,
      ),
    ).toMatchSnapshot()
  })
})
