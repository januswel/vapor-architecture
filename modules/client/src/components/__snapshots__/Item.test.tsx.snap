// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders tappable item for available item 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 168,
          "justifyContent": "space-around",
          "margin": 16,
          "width": 168,
        },
        Object {},
      ]
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "https://example.com/assets/images/coffee.png",
        }
      }
      style={
        Object {
          "height": 96,
          "width": 96,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      짜
      120
    </Text>
  </View>
</View>
`;

exports[`Item renders untappable item for unavailable item 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 168,
        "justifyContent": "space-around",
        "margin": 16,
        "width": 168,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://example.com/assets/images/coffee.png",
      }
    }
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    짜
    120
  </Text>
</View>
`;

exports[`Item renders untappable item for unavailable item 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 168,
        "justifyContent": "space-around",
        "margin": 16,
        "width": 168,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://example.com/assets/images/coffee.png",
      }
    }
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    짜
    120
  </Text>
</View>
`;

exports[`Item renders untappable item for unavailable item 3`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "height": 168,
        "justifyContent": "space-around",
        "margin": 16,
        "width": 168,
      },
      Object {
        "backgroundColor": "gray",
      },
    ]
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://example.com/assets/images/coffee.png",
      }
    }
    style={
      Object {
        "height": 96,
        "width": 96,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 16,
      }
    }
  >
    짜
    120
  </Text>
</View>
`;
